App/PyImports/ProjectSentinel.py

```diff
@@ -34,16 +35,40 @@ def loadPhases(self, phases_rcif_path):
 
         pass
 
-    @Slot(str)
-    def loadExperiment(self, experiment_rcif_path):
+    @Slot(str, str)
+    def loadExperiment(self, experiment_rcif_path, selected_name_filter):
         """
         Load an experiment information from a file.
         :param experiment_rcif_path: URI to experiment (r)cif file
         :return:
         """
+        
         self.experiment_rcif_path = self.generalizePath(experiment_rcif_path)
-
-        pass
+        
+        if "(*.cif)" in selected_name_filter:
+            self.experiment_file_format = "cif"
+        elif "(*.xye)" in selected_name_filter:
+            self.experiment_file_format = "xye"
+        
+            data = np.loadtxt(self.experiment_rcif_path)
+            joined = [" ".join(item) for item in data.astype(str)]
+            data_string = "\n".join(joined)
+            
+            n_columns_unpolarized = 3
+            n_columns_polarized = 5
+            
+            # Determine if the loaded data set is polarized or unpolarized
+            if data.shape[1] == n_columns_unpolarized:
+                cif_string = UNPOLARIZED_CIF_HEADER() + data_string
+            elif data.shape[1] == n_columns_polarized:
+                cif_string = POLARIZED_CIF_HEADER() + data_string
+            else:
+                raise IOError("Given xye file did not contain 3 or 5 columns of data.")
+                
+            self._cif_string = cif_string
+        
+        else:
+            raise IOError("Given selected_name_filter not handled in loadExperiment.")
```

Comment:
Please, add unit tests to cover the lines 64-70 in ProjectSentinel.py. See coverage on: https://codecov.io/gh/easyDiffraction/easyDiffraction/src/DAS-229/App/PyImports/ProjectSentinel.py

Reply:
Improved the coverage significantly.